src/common/cockpitwebresponse.c

```diff
@@ -1172,18 +1162,10 @@ cockpit_web_response_error (CockpitWebResponse *self,
 
   g_debug ("%s: returning error: %u %s", self->logname, code, message);
 
-  if (cockpit_web_failure_resource)
-    {
-      input = g_resources_lookup_data (cockpit_web_failure_resource, G_RESOURCE_LOOKUP_FLAGS_NONE, &error);
-      if (input == NULL)
-        {
-          g_critical ("couldn't load: %s: %s", cockpit_web_failure_resource, error->message);
-          g_error_free (error);
-        }
-    }
-
-  if (!input)
-    input = g_bytes_new_static (default_failure_template, strlen (default_failure_template));
+  extern char _binary_src_common_fail_html_start[];
+  extern char _binary_src_common_fail_html_end[];
```

Comment:
These are magic symbols created by `ld -b` then?

My gut feeling is to keep it simple, and just put fail.html as a static string here. (defining a string with `#include` is neat, but allegedly only works for C++; and `xxd` is not available in a basic build system environment).

Reply:
I went through all the possibilities:
 - `xxd` would be an annoying dependency and it wouldn't advance my goal to evict stuff from `BUILT_SOURCES`
 - static string is really hard when you want to have newlines and stuff in it, without making the entire file hard to read:
```
"<html>\n"
"  <body>\n"
"    <p>\n"
...
```
 - C++11 has "raw strings" which start with `R"(`, end with `)"` and include newlines, and `gcc` even supports them in `C`, but clang doesn't
 - `#embed` is coming, but probably not for another year or two

Reply:
Right, it's fine, just wanted to explore this a bit.